coalib/output/ConsoleInteraction.py

```diff
@@ -607,6 +609,139 @@ def print_actions(console_printer, section, actions, failed_actions):
+    :choose_action:         The action object that will be applied.
+    :param console_printer: Object to print messages on the console.
+    :param section:         Currently active section.
+    :param metadata_list:   Contains metadata for all the actions.
+    :param action_dict:     Contains the action names as keys and their
+                            references as values.
+    :param failed_actions:  A set of all actions that have failed. A failed
+                            action remains in the list until it is successfully
+                            executed.
+    :param result:          Result corresponding to the actions.
+    :param file_diff_dict:  If it is an action which applies a patch, this
+                            contains the diff of the patch to be applied to
+                            the file with filename as keys.
+    :param file_dict:       Dictionary with filename as keys and its contents
+                            as values.
+    """
+    try:
+        chosen_action.apply_from_section(result,
+                                         file_dict,
+                                         file_diff_dict,
+                                         section)
+        console_printer.print(
+            format_lines(chosen_action.SUCCESS_MESSAGE, symbol='['),
+            color=SUCCESS_COLOR)
+        failed_actions.discard(action_name)
+    except Exception as exception:  # pylint: disable=broad-except
+        logging.error('Failed to execute the action {} with error: {}.'
+                      .format(action_name, exception))
+        failed_actions.add(action_name)
+
+
+def apply_chain_action(console_printer,
+                       section,
+                       metadata_list,
+                       action_dict,
+                       failed_actions,
+                       result,
+                       file_diff_dict,
+                       file_dict):
+    """
+    Asks the user for an action and applies it.
+
+    :param console_printer: Object to print messages on the console.
+    :param section:         Currently active section.
+    :param metadata_list:   Contains metadata for all the actions.
+    :param action_dict:     Contains the action names as keys and their
+                            references as values.
+    :param failed_actions:  A set of all actions that have failed. A failed
+                            action remains in the list until it is successfully
+                            executed.
+    :param result:          Result corresponding to the actions.
+    :param file_diff_dict:  If it is an action which applies a patch, this
+                            contains the diff of the patch to be applied to
+                            the file with filename as keys.
+    :param file_dict:       Dictionary with filename as keys and its contents
+                            as values.
+    :return:                Returns a boolean value. True will be returned, if
```

Comment:
Not the best wording tbh. Make it mkre like: return true if there is the possibility of the user selecting bla bla, or false otherwise.